File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string mn, mx;
        mn = mx = strs[0];
        for(const string &s: strs) {
            mn = min(mn, s);
            mx = max(mx, s);
        }
        int ans = 0;
        int n = min(mn.size(), mx.size());
        for(int i = 0; i < n; i++) {
            if(mn[i] == mx[i]) ans = i+1;
            else break;
        }
        return mn.substr(0, ans);
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        n = min(len(i) for i in strs)
        for i in range(1,n+1)[::-1]:
            if all(s[:i] == strs[0][:i] for s in strs):
                return strs[0][:i]
        return ""